### PR TITLE
fix NoneType error

### PR DESCRIPTION
**Pull request by nickpell**
http://manage.dimagi.com/default.asp?229549

Strange that Stripe would give us `None` here

### CODE REVIEW

**Comment by nickpell** [comment]
@esoergel @proteusvacuum

File: corehq/apps/accounting/models.py [hunk 1]
```diff
@@ -2866,15 +2866,17 @@ def all_cards_serialized(self, billing_account):
             'exp_year': card.exp_year,
             'token': card.id,
             'is_autopay': card.metadata.get('auto_pay_{}'.format(billing_account.id), False),
-        } for card in self.all_cards]
+        } for card in self.all_cards if card is not None]
```
Comment (esoergel) [inline corehq/apps/accounting/models.py]: Might be more future-proof to modify the definition of `all_cards` instead.

Comment (nickpell) [inline corehq/apps/accounting/models.py]: updated

File: corehq/apps/accounting/models.py [hunk 2]
```diff
@@ -2856,7 +2856,7 @@ def _get_stripe_customer(self):
 
     @property
     def all_cards(self):
-        return self.customer.cards.data
+        return filter(lambda card: card is not None, self.customer.cards.data)
```
Comment (esoergel) [inline corehq/apps/accounting/models.py]: nit, as long as you want to filter out all falsy values, you can do
```python
filter(None, self.customer.cards.data)
```

**Comment by esoergel** [comment]
:+1: on :tractor:

**Comment by czue** [comment]
@esoergel liking the emoji game. :baseball:

**Comment by nickpell** [comment]
✅

**Pull request merged**